scipy/stats/tests/test_morestats.py

```diff
@@ -1687,35 +1672,44 @@ def test_auto(self):
         x = np.arange(0, 50) + 0.5
         y = np.arange(50, 0, -1)
         assert_equal(stats.wilcoxon(x, y),
-                     stats.wilcoxon(x, y, mode="exact"))
+                     stats.wilcoxon(x, y, method="exact"))
 
         # n <= 50: if there are zeros in d = x-y, use PermutationMethod
         pm = stats.PermutationMethod()
-        d = np.arange(0, 5)
+        d = np.arange(-2, 5)
         w, p = stats.wilcoxon(d)
         # rerunning the test gives the same results since n_resamples
         # is large enough to get deterministic results if n <= 13
         # so we do not need to use a seed. to avoid longer runtimes of the
-        # test, use n=5 only. For n=13, see test_auto_permutation_edge_case
+        # test, use n=7 only. For n=13, see test_auto_permutation_edge_case
         assert_equal((w, p), stats.wilcoxon(d, method=pm))
 
         # for larger vectors (n > 13) with ties/zeros, use asymptotic test
-        d = np.arange(0, 14)
+        d = np.arange(-5, 9)  # zero
+        w, p = stats.wilcoxon(d)
+        assert_equal((w, p), stats.wilcoxon(d, method="asymptotic"))
+
+        d[d == 0] = 1  # tie
         w, p = stats.wilcoxon(d)
         assert_equal((w, p), stats.wilcoxon(d, method="asymptotic"))
 
         # use approximation for samples > 50
         d = np.arange(1, 52)
-        assert_equal(stats.wilcoxon(d), stats.wilcoxon(d, mode="asymptotic"))
+        assert_equal(stats.wilcoxon(d), stats.wilcoxon(d, method="asymptotic"))
 
     @pytest.mark.xslow
     def test_auto_permutation_edge_case(self):
-        # n <= 50: if there are zeros in d = x-y, use PermutationMethod
-        # this is a slower test to show that results are deterministic if n=13
-        pm = stats.PermutationMethod()
-        d = np.arange(0, 13)
```

Comment:
`method='exact'` gives the same result in this case. Let's test a case that allows us to distinguish between the two methods.